File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/api/command/transfer/CompleteMoneyTransferCommand.java
package org.jmolecules.example.axonframework.domain.api.command.transfer;

import org.axonframework.modelling.command.TargetAggregateIdentifier;

public record CompleteMoneyTransferCommand(

  @TargetAggregateIdentifier
  String sourceAccountId,

  String moneyTransferId

) {

}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/domain/api/command/transfer/CompleteMoneyTransferCommand.kt
package org.jmolecules.example.axonframework.domain.api.command.transfer

import org.jmolecules.architecture.cqrs.annotation.Command
import org.jmolecules.ddd.annotation.Association
import org.jmolecules.example.axonframework.domain.api.type.AccountId
import org.jmolecules.example.axonframework.domain.api.type.MoneyTransferId

/**
 * Internal command to complete money transfer.
 */
@Command(namespace = "axon.bank", name = "CompleteMoneyTransferCommand")
data class CompleteMoneyTransferCommand(

    @Association
    val sourceAccountId: AccountId,
    val moneyTransferId: MoneyTransferId

)
